### PR TITLE
Add delete appointment

### PR DESCRIPTION
**Pull request by officialchengyud**

### CODE REVIEW

File: src/main/java/seedu/address/model/AddressBook.java [hunk 1]
```diff
@@ -54,12 +54,21 @@ public void setPersons(List<Person> persons) {
         this.persons.setPersons(persons);
     }
 
+    /**
+     * Replaces the contents of the appointment list with {@code appointments}.
+     * {@code persons} must not contain duplicate persons.
```
Comment (ararchch) [inline src/main/java/seedu/address/model/AddressBook.java:59]: I think we can change the comment here to appointments so it's more relevant

File: src/test/java/seedu/address/logic/parser/DeleteAppointmentCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private DeleteAppointmentCommandParser parser = new DeleteAppointmentCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsDeleteCommand() {
+        assertParseSuccess(parser, "1", new DeleteAppointmentCommand(INDEX_FIRST_PERSON));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "a",
```
Comment (ararchch) [inline src/test/java/seedu/address/logic/parser/DeleteAppointmentCommandParserTest.java:30]: Might be more readable to split this up into two lines

File: src/test/java/seedu/address/testutil/TypicalAppointments.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public static final Appointment APPOINTMENT_2 = new AppointmentBuilder().withDate("2024-03-20")
+            .withDoctor((Doctor) BROWN).withPatient((Patient) ALICE).build();
+    public static final Appointment APPOINTMENT_3 = new AppointmentBuilder().withDate("2024-03-25")
+            .withDoctor((Doctor) BROWN).withPatient((Patient) ALICE).build();
+    private TypicalAppointments() {} // prevents instantiation
+
+    /**
+     * Returns an {@code AddressBook} with all the typical appointments.
+     */
+    public static AddressBook getTypicalAddressBook() {
```
Comment (ararchch) [inline src/test/java/seedu/address/testutil/TypicalAppointments.java:31]: Does this work without having added the persons involved in the appointment to the address book first? If it does, is there perhaps a bug that we should fix

**Review by ararchch — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return targetIndex.equals(otherDeleteCommand.targetIndex);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("targetIndex", targetIndex)
+                .toString();
+    }
+}
```
Comment (alfaloo) [inline src/main/java/seedu/address/logic/commands/DeleteAppointmentCommand.java:72]: looks amazing chengyu

File: src/test/java/seedu/address/logic/commands/DeleteAppointmentCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @Test
+    public void toStringMethod() {
+        Index targetIndex = Index.fromOneBased(1);
+        DeleteAppointmentCommand deleteAppointmentCommand = new DeleteAppointmentCommand(targetIndex);
+        String expected = DeleteAppointmentCommand.class.getCanonicalName() + "{targetIndex=" + targetIndex + "}";
+        assertEquals(expected, deleteAppointmentCommand.toString());
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/logic/commands/DeleteAppointmentCommandTest.java:79]: tests are goated

File: src/main/java/seedu/address/model/AddressBook.java (hunk 1 above)
Comment (officialchengyud) [reply src/main/java/seedu/address/model/AddressBook.java:59]: Thanks for spotting this! Making changes now.

File: src/test/java/seedu/address/logic/parser/DeleteAppointmentCommandParserTest.java (hunk 2 above)
Comment (officialchengyud) [reply src/test/java/seedu/address/logic/parser/DeleteAppointmentCommandParserTest.java:30]: I agree. Thank you for spotting this.

**Pull request merged**